documentation/book/con-custom-resources-status.adoc

```diff
@@ -6,16 +6,53 @@
 
 = {ProductName} custom resource status
 
-The `status` property of a {ProductName}-specific custom resource publishes the _current_ state of the resource to users and tools that need the information.
+The `status` property of a {ProductName} custom resource publishes useful information about the resource to users and tools that need it.
 
-Status information is useful for tracking progress related to a resource achieving its _desired_ state, as defined by the `spec` property. The `status` provides the time and reason the state of the resource changed and details of events preventing or delaying the Operator from realizing the desired state.
+Several resources have a `status` property, as described in the following table.
+
+[cols="2*",options="header",stripes="none",separator=¦]
+|===
+
+¦{ProductName} Resource
+¦Publishes information on...
+
+m¦xref:type-KafkaStatus-reference[]
+¦The Kafka cluster.
+
+m¦xref:type-KafkaTopicStatus-reference[]
+¦Kafka topics and their partitions.
+
+m¦xref:type-KafkaConnectStatus-reference[]
+¦The Kafka Connect cluster, if deployed.
+
+m¦xref:type-KafkaConnectS2Istatus-reference[]
+¦The Kafka Connect cluster with Source-to-Image support, if deployed.
+
+m¦xref:type-KafkaMirrorMakerStatus-reference[]
+¦The Kafka MirrorMaker tool, if deployed.
+
+m¦xref:type-KafkaUserStatus-reference[]
+¦Kafka users.
+
+m¦xref:type-KafkaBridgeStatus-reference[]
+¦The {ProductName} Kafka Bridge, if deployed.
+
+|===
+
+The `status` property includes information on:
+
+* The _current state_ of the resource described in _status conditions_
+
+* The resource's last `observedGeneration` (the generation of the resource to which the status applies)
```

Comment:
Should we explain a bit more how does this related to the `generation` field in the `metadata` section?

Reply:
I can't find the any docs on the `generation` field in the API Reference. Is it missing?

Reply:
It is a Kubernetes field, so it is not in our API reference. Basically, the `metadata.generation` field is an integer which is incremented everytime the resource is modified (i.e. the user changes it with `kubectl apply`, `kubectl edit` etc.). The `status.observedGeneration` basically mirrors the `metadata.generation` field to show what was the last generation of this resource which the operator has already seen.

The usefulness for the user is that the user can compare these number and when they are the same it means that the operator saw the latest version and the status is valid for it. If the `observedGeneration` is smaller then the `generation`, it means that the operator did not process the last update to the custom resource yet.

I didn't find any link in Kubernetes docs either - apart from the API reference which doesn't explain much. So I think it would be good to add a bit from what I described here into the docs.

Reply:
Thanks for this. I've added a brief description of "last observed generation".